### ISSUE TITLE
surface=erde is gone from OSM tagging

### ISSUE THREAD

**Issue by matkoniecz** [vmarc/knooppuntnet#397]
In this case `surface=erde` is already replaced by other more established tagging (likely `surface=dirt` or `surface=earth` if it was accurate in the first place)

In this case I systematically created notes in places where `surface=erde` was used, this notes were processed and tag is now gone.

Are you interested in getting such issues at all? It was suggested to me to create such issues in cases where my activity contributed to tag being eliminated - for cases where taginfo lists projects processing it.

Would you prefer to not get such contact attempts? In such case feel free to ignore it (or just close it) and I will not send any like that.

Would you prefer to get such issue when I start considering replacement of one tag by another as a clear alias?

Or only once bot edit is officially proposed for given tag?

Or only once bot edit replacing such tag with another is running?

Or only once tag is fully gone?

Would it be different if some tag is replaced only manually without proposing bot edit? (like here) Or only where there is an approved bot edit to regularly replace any new occurrences?

Would you prefer separate issue for each tag? Update existing open one with additional listings? Update existing one even if it is closed? No issues at all?

(note: tag being gone now, even if approved reoccurring bot edit does not mean it will never reappear)

**Comment by vmarc** [comment]
> Are you interested in getting such issues at all?

Issues like this are certainly very welcome. 

If there is logic in the knooppuntnet code that no longer makes sense, it is best that it gets cleaned up.

> Would you prefer to get such issue when I start considering replacement of one tag by another as a clear alias?
> 
> Or only once bot edit is officially proposed for given tag?
> 
> Or only once bot edit replacing such tag with another is running?
> 
> Or only once tag is fully gone?

I think it is worth while raising the issue as soon as you think a tag should no longer be interpreted by knooppuntnet. In the specific case of the surface tag, when a surface value is not interpreted anymore, this will result in the way being display in blue in the planner indicating surface "unknown".  This may help in signaling ways where the surface tag should be investigated by a mapper.

The logic for surface analysis is in [SurfaceAnalyzer](https://github.com/vmarc/knooppuntnet/blob/develop/server/src/main/scala/kpn/server/analyzer/engine/analysis/route/segment/SurfaceAnalyzer.scala).

I suppose following list contains other surface values that should be removed:

```
    "shell",
    "shells",
    "ash",
    "bad",
    "clay",
    "cob",
    "compact",
    "erde",    <---
    "gr",
    "loam",
    "peb",
    "soil",
    "shotter",
    "rock",
    "turf"
```

**Comment by matkoniecz** [comment]
> I suppose following list contains other surface values that should be removed:

Yes, for example `surface=shotter` has zero uses worldwide: http://overpass-turbo.eu/s/1JuA

(are you using also `cycleway:surface` and other tags in surface family?)

> when a surface value is not interpreted anymore, this will result in the way being display in blue in the planner indicating surface "unknown". This may help in signaling ways where the surface tag should be investigated by a mapper.

I guess it depends on design goals, but say `surface=gr` is never correct - is it `grass`? `gravel`? But either is unpaved so this much maybe can be assumed...

Lets take `compact` - that is design part whether tool wants to show it as suspect or alias it to `surface=compacted`. How much focus should be on QA and fixing data and how much on extracting as much as possible from badly tagged objects?

**Comment by vmarc** [comment]
> How much focus should be on QA and fixing data and how much on extracting as much as possible from badly tagged objects?

Indeed, that is a key question. Like the comment in the code says: the quoted list contains _"extra entries on top of wiki, based on openfietsmap"_. The idea at the time was to interprete as much as possible all surface values that existed in the wild even if these values were not documented on the wiki.  Today I think I would prefer badly tagged objects to be flagged (displayed as _"surface unknown"_) rather than quessing the meaning (so more QA oriented).

> (are you using also cycleway:surface and other tags in surface family?)

Tag _"cycleway:surface"_ is interpreted and takes precedence over _"surface"_ if present, both in hiking and cycling networks. In hiking networks _"footway:surface"_ takes precedence over _"cycleway:surface"_. See line 153 _preferredSurfaceTagKey()_ in the code.

**Comment by matkoniecz** [comment]
> Tag "cycleway:surface" is interpreted

Then I guess that this also should be checked whether such values going extinct are present there (unless QA-style behaviour is preferable anyway)

**Comment by vmarc** [comment]
Removed tag values from surface analysis (paved/unpaved) logic:

- erde

- compact

- peb

- gr

- shotter

**Comment by matkoniecz** [comment]
https://taginfo.openstreetmap.org/tags/surface=compact#projects still lists it

https://taginfo.openstreetmap.org/projects/knooppuntnet#overview lists https://raw.githubusercontent.com/vmarc/knooppuntnet/develop/docs/taginfo.json as source and it is still there

maybe this file needs to be regenerated? Or is being updated manually
?

Or maybe I should wait for a release?

**Comment by vmarc** [comment]
Yes, [taginfo.json](https://raw.githubusercontent.com/vmarc/knooppuntnet/develop/docs/taginfo.json) needed to be updated by running [TagInfoReport](https://github.com/vmarc/knooppuntnet/blob/develop/server/src/main/scala/kpn/core/taginfo/TagInfoReport.scala). Just did that.

**Comment by matkoniecz** [comment]
What about `surface=bad` or `surface=paving_stone` ? Seems still listed by taginfo.
